test/models/host_test.rb

```diff
@@ -678,45 +678,42 @@ def teardown
     refute_equal original_status, new_status
   end
 
-  test "assign a host to a location" do
-    host = Host.create :name => "host 1", :mac => "aabbecddeeff", :ip => "5.5.5.5", :hostgroup => hostgroups(:common), :managed => false
-    location = Location.create :name => "New York"
-
-    host.location_id = location.id
-    assert host.save!
-  end
-
-  test "update a host's location" do
-    host = Host.create :name => "host 1", :mac => "aabbccddeeff", :ip => "5.5.5.5", :hostgroup => hostgroups(:common), :managed => false
-    original_location = Location.create :name => "New York"
+  context 'host assigned to location and organization' do
+    setup do
+      @host = Host.create :name => "host 1", :mac => "aabbecddeeff", :ip => "5.5.5.5", :hostgroup => hostgroups(:common), :managed => false
+      @location = Location.create :name => "New York"
+      @organization = Organization.create :name => "Hosting client 1"
+    end
 
-    host.location_id = original_location.id
-    assert host.save!
-    assert host.location_id = original_location.id
+    test "assign a host to a location" do
+      @host.location_id = @location.id
+      assert @host.save!
+    end
 
-    new_location = Location.create :name => "Los Angeles"
-    host.location_id = new_location.id
-    assert host.save!
-    assert host.location_id = new_location.id
-  end
+    test "update a host's location" do
+      original_location = @location
 
-  test "assign a host to an organization" do
-    host = Host.create :name => "host 1", :mac => "aabbecddeeff", :ip => "5.5.5.5", :hostgroup => hostgroups(:common), :managed => false
-    organization = Organization.create :name => "Hosting client 1"
+      @host.location_id = original_location.id
+      assert @host.save!
+      assert @host.location_id = original_location.id
```

Comment:
I think this should be `assert_equal` - looks like this test was wrong a long time (it has an assignment instead of a comparison)